modules/openapi-generator/src/main/resources/dart2/api_client.mustache

```diff
@@ -1,11 +1,13 @@
 {{>header}}
 {{>part_of}}
-class ApiClient {
-  ApiClient({this.basePath = '{{{basePath}}}', this.authentication});
+class ApiClient<T extends Authentication> {
```

Comment:
Why is this change necessary? even if the developer inherits `ApiClient`, there is no real use for the generic parameter

Reply:
```dart
abstract CustomAuthenticationMechanism extends Authentication {
  Future<String?> get webTokenProvider;
}

class CustomApiClient extends ApiClient<CustomAuthenticationMechanism> {
  // We have access to authentication.webTokenProvider
}
```

Reply:
why not :
```dart
class CustomApiClient extends ApiClient {
  @override
  final CustomAuthenticationMechanism? authentication;
}
```

Reply:
Looks promising, does it work? 🙂 I'll check it and if works then I'll revert. Thanks!

Reply:
Looks like it works, so reverted addition of `T` generic. Thanks for the tip!